dialogs.c

```diff
@@ -436,6 +463,36 @@ static void refresh_usb(void)
 	gtk_combo_box_set_active(GTK_COMBO_BOX(dialogs.connect_usbd), index);
 }
 
+static void refresh_serial(void) {
+	GtkListStore *liststore;
+	struct sp_port **ports;
+	gchar *buf;
+	int i;
+
+	/* clear everything, and scan again */
+	liststore = GTK_LIST_STORE(gtk_combo_box_get_model(GTK_COMBO_BOX(dialogs.connect_seriald)));
+	gtk_list_store_clear(liststore);
+
+	/*get serial ports*/
+	enum sp_return error = sp_list_ports(&ports);
+	if (error == SP_OK) {
+	for (i = 0; ports[i]; i++) {
+		buf = malloc(strlen(sp_get_port_name(ports[i])) + 5);
+		sprintf(buf, "%s ",  sp_get_port_name(ports[i]));
```

Comment:
Just for my understanding, why the extra space at the end?

Reply:
This is a mistake from my part. Sorry for that,  it also eliminates the removal of that space when creating the URI. 
port[strlen(port) - 1] = '\0';